### PR TITLE
Proposed fix for swayidle forking behaviour

### LINKED ISSUE #3119: After resuming from suspend: desktop briefly shows up before lockscreen
* Sway version: 1.0-beta.1-174-g80a1c340 (Nov 11 2018, branch 'master')
* Debug log: https://ptpb.pw/lDoA
* Config file: https://ptpb.pw/nAMb
The config file is the default one with just the swayidle example commented in, plus different $mod and $term.

**Steps to reproduce**

Start sway, open terminal and issue 'systemctl suspend', press key for resume.
What happens: I see my desktop for 0.6 seconds, then I see the lockscreen.
What should happen: I see the lockscreen.

Unlocking works fine btw.
This is on a Samsung Chromebook Plus, running ArchlinuxARM: Linux alarm 4.19.1-2-ARCH #1 SMP Sun Nov 4 19:18:32 MST 2018 aarch64 GNU/Linux

### PR DESCRIPTION
**Pull request by ianyfan**
This should fix #3119, but I'm not sure whether it has the potential to leave zombie processes.

### CODE REVIEW

**Comment by ianyfan** [comment]
I don't think forking should be the default, but if you want it to be, please suggest what the flag and description should be.

**Comment by emersion** [comment]
I think `-f` is confusing. `swaylock -f` _daemonizes_ swaylock (ie. swaylock starts in the background). Here, the behaviour is different: swayidle either waits for child processes to finish, or runs them in the background.

**Comment by ianyfan** [comment]
Makes sense. I've changed it to `-w` to wait for the command to finish executing. What do you think?

**Comment by emersion** [comment]
This looks good. I think the example config still needs to be updated, to use `swayidle -w` and `swaylock -f`?

**Comment by emersion** [comment]
We also need to make sure this correctly locks the screen before idling.

File: swayidle/main.c [hunk 1]
```diff
@@ -97,29 +90,25 @@ static void acquire_sleep_lock(void) {
...
+		goto cleanup;
 	}
 
 	// sd_bus_message_unref closes the file descriptor so we need
 	// to copy it beforehand
 	lock_fd = fcntl(lock_fd, F_DUPFD_CLOEXEC, 3);
 	if (lock_fd < 0) {
-		wlr_log(WLR_ERROR, "Failed to copy sleep lock fd: %s",
-			strerror(errno));
+		wlr_log(WLR_ERROR, "Failed to copy sleep lock fd: %s", strerror(errno));
```
Comment (emersion) [inline swayidle/main.c]: Missing `goto` here

File: swayidle/swayidle.1.scd [hunk 2]
```diff
@@ -16,6 +16,10 @@ swayidle - Idle manager for Wayland
 *-d*
 	Enable debug output.
 
+*-w*
+	Wait for command to finish executing before continuing, helpful for ensuring
+	that a *before-sleep* command has finished before the system goes to sleep.
```
Comment (emersion) [inline swayidle/swayidle.1.scd]: Maybe add a warning like "swayidle will block until the command finishes".

**Pull request closed without merging**